src/compiler/checker.ts

```diff
@@ -5182,7 +5220,14 @@ namespace ts {
         }
 
         function isContextSensitiveFunctionLikeDeclaration(node: FunctionLikeDeclaration) {
-            return !node.typeParameters && node.parameters.length && !forEach(node.parameters, p => p.type);
+            if (compilerOptions.strictThis) {

```

Comment:
Why this check? It doesn't matter if were in strict-this mode, we *always* want to ignore `this` parameters for purposes of this function.